src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,35 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+	public final String value;
+	public boolean isPrivate;
+	
+	public Contact(String value, boolean isPrivate){
+		this.value = value;
+		this.isPrivate = isPrivate;
+	}
+	
+
+	
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Email // instanceof handles nulls
+                && this.value.equals(((Email) other).value)); // state check
```

Comment:
Using child class(Email) in parent class(Contact) will cause error. The parent class shouldn't know about its children classes. You should implement the method `equals` in the children class individually.